src/app/base/components/SecondaryNavigation/SecondaryNavigation.tsx

```diff
@@ -0,0 +1,211 @@
+/* eslint-disable react/no-multi-comp */
+import classNames from "classnames";
+import type { Location } from "react-router-dom-v5-compat";
+import { Link, matchPath, useLocation } from "react-router-dom-v5-compat";
+
+import settingsURLs from "app/settings/urls";
+
+export type NavItem = {
+  label: string;
+  path?: string;
+  items?: NavItem[];
+};
+
+type ItemProps = { item: NavItem };
+
+const SideNavigationLink = ({ item }: ItemProps) => {
+  const location = useLocation();
+  const isActive = getIsActive({ item, location });
+  if (!item.path) {
+    return null;
+  }
+  return (
+    <Link
+      aria-current={isActive ? "page" : undefined}
+      className={classNames("p-side-navigation__link", {
+        "is-active": isActive,
+      })}
+      to={item.path}
+    >
+      {item.label}
+    </Link>
+  );
+};
+
+const SubNavItem = ({ item }: ItemProps) => {
+  return (
+    <li className="p-side-navigation__item" key={item.path}>
+      <SideNavigationLink item={item} />
+    </li>
+  );
+};
+
+const SubNavigation = ({ items }: { items: NavItem["items"] }) => {
+  if (!items || !items.length) return null;
+
+  return (
+    <ul className="p-side-navigation__list">
+      {items.map((item) => (
+        <SubNavItem item={item} key={item.path} />
+      ))}
+    </ul>
+  );
+};
+
+const getIsActive = ({
+  item,
+  location,
+}: ItemProps & { location: Location }) => {
+  const path = item.path;
+
+  if (!path) {
+    return false;
+  }
+
+  if (!item.items) {
+    return location.pathname.startsWith(path);
+  }
+
+  return matchPath(path, location.pathname);
+};
+
+const SideNavigationItem = ({ item }: ItemProps) => {
```

Comment:
Make sure you check all places where `SideNav` component is used. For example, the preferences page `MAAS/r/account/prefs/details`  doesn't have the sidebar displayed at all.